File: Code_VB/C15_UserControls/CustomControls/CountedButton.vb
Imports System 
Imports System.ComponentModel 
Imports System.Web.UI 
Imports System.Web.UI.WebControls 

Namespace CustomControls 
    <DefaultProperty("Text")> _ 
    <ToolboxData("<{0}:CountedButton runat=server></{0}:CountedButton>")> _ 
    Public Class CountedButton 
        Inherits Button 
        Public Sub New() 
            Text = "Click me" 
            ViewState("Count") = 0 
        End Sub 
        
        Public Property Count() As Integer 
            Get 
                Return Convert.ToInt32(ViewState("Count")) 
            End Get 
            Set(ByVal value As Integer) 
                ViewState("Count") = value 
            End Set 
        End Property 
        
        Protected Overloads Overrides Sub OnClick(ByVal e As EventArgs) 
            Count = Count + 1 
            Text = Count.ToString() & " clicks" 
            MyBase.OnClick(e) 
        End Sub 
    End Class 
End Namespace 

File: Code_CS/C15_UserControls/CustomControls/CountedButton.cs
using System;
using System.ComponentModel;
using System.Web.UI;
using System.Web.UI.WebControls;

namespace CustomControls
{
    [DefaultProperty("Text")]
    [ToolboxData("<{0}:CountedButton runat=server></{0}:CountedButton>")]
    public class CountedButton : Button
    {
        public CountedButton()
        {
            Text = "Click me";
            ViewState["Count"] = 0;
        }

        public int Count
        {
            get { return Convert.ToInt32(ViewState["Count"]); }
            set { ViewState["Count"] = value; }
        }

        protected override void OnClick(EventArgs e)
        {
            Count = Count + 1;
            Text = Count.ToString() + " clicks";
            base.OnClick(e);
        }
    }
}
